crates/uv-publish/src/trusted_publishing.rs

```diff
@@ -94,34 +79,39 @@ pub struct OidcTokenClaims {
 }
 
 /// Returns the short-lived token to use for uploading.
+///
+/// Return states:
+/// - `Ok(Some(token))`: Successfully obtained a trusted publishing token.
+/// - `Ok(None)`: Not in a supported CI environment for trusted publishing.
+/// - `Err(...)`: An error occurred while trying to obtain the token.
 pub(crate) async fn get_token(
     registry: &DisplaySafeUrl,
     client: &ClientWithMiddleware,
-) -> Result<TrustedPublishingToken, TrustedPublishingError> {
-    // If this fails, we can skip the audience request.
-    let oidc_token_request_token =
-        env::var(EnvVars::ACTIONS_ID_TOKEN_REQUEST_TOKEN).map_err(|err| {
-            TrustedPublishingError::from_var_err(EnvVars::ACTIONS_ID_TOKEN_REQUEST_TOKEN, err)
-        })?;
-
-    // Request 1: Get the audience
+) -> Result<Option<TrustedPublishingToken>, TrustedPublishingError> {
+    // Get the OIDC token's audience from the registry.
     let audience = get_audience(registry, client).await?;
 
-    // Request 2: Get the OIDC token from GitHub.
-    let oidc_token = get_oidc_token(&audience, &oidc_token_request_token, client).await?;
+    // Perform ambient OIDC token discovery.
+    // Depending on the host (GitHub Actions, GitLab CI, etc.)
+    // this may perform additional network requests.
+    let oidc_token = get_oidc_token(&audience, client).await?;
 
-    // Request 3: Get the publishing token from PyPI.
-    let publish_token = get_publish_token(registry, &oidc_token, client).await?;
+    // Exchange the OIDC token for a short-lived upload token,
+    // if OIDC token discovery succeeded.
+    if let Some(oidc_token) = &oidc_token {
+        let publish_token = get_publish_token(registry, oidc_token.reveal(), client).await?;
```

Comment:
nit: We can pass an id token as argument and only reveal it for building the json payload.